ds3-autogen-c/src/test/java/com/spectralogic/ds3autogen/c/RequestConverter_Test.java

```diff
@@ -63,4 +63,54 @@ public void testGetRequestParameterListWithResponseType() {
         assertEquals(paramList.size(), 3);
         assertEquals(RequestHelper.paramListToString(paramList), "const ds3_client* client, const ds3_request* request, const ds3_get_service_response** response");
     }
+
+    @Test
+    public void testConvertDs3RequestWithRequestPayload() {
+        final Request testRequest = RequestConverter.toRequest(
+                new Ds3Request("CreateGetJobRequestHandler",
+                        HttpVerb.PUT,
+                        Classification.spectrads3,
+                        null, // bucketRequirement
+                        null, // objectRequirement
+                        Action.MODIFY, // Action
+                        Resource.BUCKET, //Resource
+                        ResourceType.NON_SINGLETON, // ResourceType
+                        Operation.START_BULK_GET, // Operation
+                        true, // includeIdInPath
+                        null, // ds3ResponseCodes,
+                        ImmutableList.of(
+                                new Ds3Param("Aggregating", "boolean", false),
+                                new Ds3Param("ChunkClientProcessingOrderGuarantee", "com.spectralogic.s3.common.dao.domain.ds3.JobChunkClientProcessingOrderGuarantee", false),
+                                new Ds3Param("Name", "java.lang.String", true),
+                                new Ds3Param("Priority", "com.spectralogic.s3.common.dao.domain.ds3.BlobStoreTaskPriority", false)), // optionalQueryParams
+                        ImmutableList.of(
+                                new Ds3Param("Operation", "com.spectralogic.s3.server.request.rest.RestOperationType", true))));// requiredQueryParams
+        assertTrue(testRequest.hasRequestPayload());
+    }
+
+    @Test
+    public void testNullResponseForNoRequestPayload() {
+        final ImmutableMap<String, Parameter> requestPayloadMap = RequestConverter.buildRequestPayloadMap();
+        assertEquals(requestPayloadMap.get("unknownRequest"), null);
+    }
+
+    @Test
+    public void testConvertDs3RequestWithoutRequestPayload() {
+        final Request testRequest = RequestConverter.toRequest(
+                new Ds3Request("CreateBucketRequestHandler",
```

Comment:
This should use the new fixture function createBucketRequest() instead of reconstructing the Ds3 model